src/lib/strategy-card/leadership.js

```diff
@@ -0,0 +1,82 @@
+const {
+    onUiClosedClicked,
+    createStrategyCardUi,
+    broadcastMessage,
+} = require("./strategy-card");
+const {
+    globalEvents,
+    Button,
+    CheckBox,
+    Slider,
+    Text,
+    VerticalBox,
+} = require("../../wrapper/api");
+const locale = require("../../lib/locale");
+
+let selections = {};
+let activatingPlayer;
+
+function getPlayerSelectionBySlot(player) {
+    const slot = player.getSlot();
+    selections[slot] = selections[slot] || {
+        value: 0,
+        primary: false,
+    };
+
+    return selections[slot];
+}
+
+function createUiWidgetFactory() {
+    let primaryCheckBox = new CheckBox()
+        .setFontSize(10)
+        .setText(locale("strategy_card.leadership.text.primary"));
+    primaryCheckBox.onCheckStateChanged.add((checkBox, player, isChecked) => {
+        getPlayerSelectionBySlot(player).primary = isChecked;
+    });
+    let slider = new Slider().setStepSize(1).setMaxValue(10);
```

Comment:
I personally like the numbered buttons for simple click, curious to see how a slider feels.

Reply:
I liked the idea of having the "primary" checkbox and the slider to align with my thinking: "i gain 3 from my 9 influence and played the primary ability" - I check the feedback and we will see how it is recieved